earthpy/tests/test_stack.py

```diff
@@ -0,0 +1,98 @@
+""" Tests for the stack() method """
+
+import os
+import numpy as np
+import pytest
+import earthpy.spatial as es
+
+
+def test_stack_no_file_ext(basic_image_tif):
+    """Test for error raised when no file extension provided in output file."""
+
+    # Create list of 4 basic_image_tif files (filepaths)
+    band_files = [basic_image_tif] * 4
+
+    # Test that out_path needs a file extension to be valid
+    out_fi = "test_stack"
+    with pytest.raises(
+        ValueError, match="Please specify a valid file name for output."
+    ):
+        stack_arr, stack_prof = es.stack(band_files, out_path=out_fi)
+
+
+def test_stack_yes_file_ext(basic_image_tif):
+    """Test for valid file extension."""
+
+    # Test that out_path needs a file extension to be valid
+    out_fi = "test_stack.tif"
+    with pytest.raises(ValueError, match="The list of"):
+        stack_arr, stack_prof = es.stack([], out_path=out_fi)
+
+
+def test_stack_out_format(basic_image_tif):
+    """Test that output format matches input format."""
+
+    # Create list of 4 basic_image_tif files (filepaths)
+    band_files = [basic_image_tif] * 4
+
+    # Test that the output file format is same as inputs
+    # This can be flexible but for now forcing the same format
+    out_fi = "test_stack.jp2"
+    with pytest.raises(ValueError, match="Source"):
+        stack_arr, stack_prof = es.stack(band_files, out_path=out_fi)
+
+
+def test_stack_outputfile(basic_image_tif):
+    """Test successful creation of output file."""
+
+    # Create list of 4 basic_image_tif files (filepaths)
+    band_files = [basic_image_tif] * 4
+
+    # Test valid use case specifying output file.
+    # Make sure the output file exists and then clean it up
+    out_fi = "test_stack.tif"
+    stack_arr, stack_prof = es.stack(band_files, out_path=out_fi)
+
+    assert os.path.exists(out_fi)
+    if os.path.exists(out_fi):
+        os.remove(out_fi)
+
+
+def test_stack_return_array(basic_image_tif):
+    """ Test returning only array."""
+
+    # Create list of 4 basic_image_tif files (filepaths)
+    band_files = [basic_image_tif] * 4
+
+    # Test valid use case of just getting back the array.
+    stack_arr, stack_prof = es.stack(band_files)
+
+    assert stack_arr.shape[0] == len(band_files)
+    assert stack_prof["count"] == len(band_files)
+
+
+def test_stack_nodata(basic_image_tif):
+    """Test nodata parameter for masking stacked array."""
+
+    # Create list of 4 basic_image_tif files (filepaths)
+    band_files = [basic_image_tif] * 4
+
+    # Test the nodata parameter
+    stack_arr, stack_prof = es.stack(band_files, nodata=0)
+
+    assert 0 not in stack_arr
```

Comment:
This is great to make sure that the 0 values were masked out. We can make it even more specific by verifying that the values that were not supposed to be masked weren't accidentally masked (for example, if we were to accidentally mask all of the values, this test would still pass).

For example, in `basic_image`, 9 of 100 cells have the value `1`, and the rest are `0`, so we should have masked 91% of the cells:

```suggestion
    assert 0 not in stack_arr and np.mean(stack_arr.mask) == 0.91
```

Reply:
sure. By using `np.ma.masked_equal()`, though, it seems redundant. I'll add this in though and make sure the test passes!

Reply:
one note here... I have this implemented now but np.sum() on the array before masking, and after masking, come out to be the same since the array contains only values 0 or 1. We can leave it as-is, but a more specific test would be to pepper in some different values and mask on something that isn't 0.